certbot-apache/certbot_apache/configurator.py

```diff
@@ -959,9 +968,100 @@ def _copy_create_ssl_vhost_skeleton(self, avail_fp, ssl_fp):
+                    vhost.filep, ssl_fp), reporter.MEDIUM_PRIORITY)
+        self.aug.set("/augeas/files%s/mtime" % (self._escape(ssl_fp)), "0")
+        self.aug.set("/augeas/files%s/mtime" % (self._escape(vhost.filep)), "0")
+
+    def _prepare_ssl_skeleton_contents(self, contents):
+        """ Helper function for _copy_create_ssl_vhost_skeleton to prepare the
+        new HTTPS VirtualHost contents. Currently disabling the rewrites """
+
+        result = []
+        sift = False
+        contents = iter(contents)
+
+        comment = ("# Some rewrite rules in this file were "
+                   "disabled on your HTTPS site,\n"
+                   "# because they have the potential to create "
+                   "redirection loops.\n")
+
+        for line in contents:
+            A = line.lower().lstrip().startswith("rewritecond")
+            B = line.lower().lstrip().startswith("rewriterule")
+
+            if not (A or B):
+                result.append(line)
+                continue
+
+            # A RewriteRule that doesn't need filtering
+            if B and not self._sift_rewrite_rule(line):
+                result.append(line)
+                continue
+
+            # A RewriteRule that does need filtering
+            if B and self._sift_rewrite_rule(line):
+                if not sift:
+                    result.append(comment)
+                    sift = True
+                result.append("# " + line)
+                continue
+
+            # We save RewriteCond(s) and their corresponding
+            # RewriteRule in 'chunk'.
+            # We then decide whether we comment out the entire
+            # chunk based on its RewriteRule.
+            chunk = []
+            if A:
+                chunk.append(line)
+                line = next(contents)
+
+                # RewriteCond(s) must be followed by one RewriteRule
+                while not line.lower().lstrip().startswith("rewriterule"):
+                    chunk.append(line)
+                    line = next(contents)
+
+                # Now, current line must start with a RewriteRule
+                chunk.append(line)
+
+                if self._sift_rewrite_rule(line):
+                    if not sift:
+                        result.append(comment)
+                        sift = True
+
+                    result.append('\n'.join(
+                        ['# ' + l for l in chunk]))
+                    continue
+                else:
+                    result.append('\n'.join(chunk))
+                    continue
+        return result, sift
+
+    def _get_vhost_block(self, vhost):
+        """ Helper method to get VirtualHost contents from the original file.
+        This is done with help of augeas span, which returns the span start and
+        end positions
+
+        :returns: `list` of VirtualHost block content lines without closing tag
+        """
+
+        try:
+            span_val = self.aug.span(vhost.path)
```

Comment:
mod_macro is the only directive I can think of that this would break. However we are specifically NOT supporting it at the moment, and supporting it would need a more fine tuned directive extraction anyway. 

From Apache perspective, the best way to handle this would be to put the newly created VirtualHosts right before or after the already existing ones in the original files, it would make these cases non-issues as well.. Distribution specific configuration schemes and us being as transparent and intuitive as possible disagree with this however.